extensions/amp-dailymotion/dailymotion-api.js

```diff
@@ -0,0 +1,98 @@
+import {addParamsToUrl} from '../../src/url';
+import {dict} from '#core/types/object';
+import {isAutoplaySupported} from '#core/dom/video';
+
+/**
+ *
+ * @param {!Window} win
+ * @param {string} videoId
+ * @param {boolean|undefined} autoplay
+ * @param {boolean|undefined} endscreenEnable
+ * @param {boolean|undefined} info
+ * @param {boolean|undefined} mute
+ * @param {boolean|undefined} sharingEnable
+ * @param {string|undefined} start
+ * @param {string|undefined} uiHighlight
+ * @param {boolean|undefined} uiLogo
+ * @param {!JsonObject} implicitParams
+ * @return {string}
+ */
+export function getDailymotionIframeSrc(
+  win,
+  videoId,
+  autoplay = false,
+  endscreenEnable = true,
+  info = true,
+  mute = false,
+  sharingEnable = true,
+  start,
+  uiHighlight,
+  uiLogo = true,
+  implicitParams
+) {
+  return addParamsToUrl(
+    `https://www.dailymotion.com/embed/video/${encodeURIComponent(
+      videoId
+    )}?api=1&html=1&app=amp`,
+    Object.assign(
+      dict({
+        // In the dailymotion API endscreenEnable, info, sharingEnable, and uiLogo
+        // all default to true, so if the attr is not marked as false do not add it to the URL
+        'endscreen-enable':
+          endscreenEnable !== true ? endscreenEnable : undefined,
+        'info': info !== true ? info : undefined,
+        // In order to support autoplay the video needs to be muted on load so we
+        // dont receive an unmute event which prevents the video from autoplay.
+        'mute': mute || (autoplay && isAutoplaySupported(win)) ? 1 : undefined,
+        'sharing-enable': sharingEnable !== true ? sharingEnable : undefined,
+        'start': start,
+        'ui-highlight': uiHighlight,
+        'ui-logo': uiLogo !== true ? uiLogo : undefined,
```

Comment:
nit: If these are real booleans we don't need triple equality.

```suggestion
        // In the dailymotion API endscreenEnable, info, sharingEnable, and uiLogo
        // all default to true, so if the attr is not marked as false do not add it to the URL
        'endscreen-enable': !endscreenEnable ? endscreenEnable : undefined,
        'info': !info ? info : undefined,
        // In order to support autoplay the video needs to be muted on load so we
        // dont receive an unmute event which prevents the video from autoplay.
        'mute': mute || (autoplay && isAutoplaySupported(win)) ? 1 : undefined,
        'sharing-enable': !sharingEnable ? sharingEnable : undefined,
        'start': start,
        'ui-highlight': uiHighlight,
        'ui-logo': !uiLogo ? uiLogo : undefined,
```